src/hunit/service/backend/rest/rest.go

```diff
@@ -61,10 +65,38 @@ func New(conf service.Config) (service.Service, error) {
 		}
 	}
 
+	entityHandler := func(endpoints []Endpoint) router.Handler {
+		return func(req *router.Request, cxt router.Context) (*router.Response, error) {
+			for _, e := range endpoints {
+				if routeMatches(e, req) {
+					bodyMatch, err := bodyMatches(e.Request.Entity, req)
+					if err != nil {
+						return nil, err
+					}
+
+					if bodyMatch {
+						return handleRequest((*http.Request)(req), cxt, e, vars)
+					}
+				}
+			}
+			return nil, fmt.Errorf("%s * * * Could not find matching route based on request body: %v: %v\n", prefix, req.URL, err)
+		}
+	}
+
 	r := router.New()
+
 	for _, e := range suite.Endpoints {
 		if e.Request != nil {
-			r.Add(e.Request.Path, handler(e)).Methods(e.Request.Methods...).Params(convertParams(e.Request.Params))
+			var routeHandler router.Handler
+
+			// if endpoint has request entity defined, use entityHandler
+			if e.Request.Entity != "" {
+				routeHandler = entityHandler(suite.Endpoints)
+			} else {
+				routeHandler = handler(e)
+			}
```

Comment:
It occurs to me that `entityHandler()` always takes the same parameter here (as opposed to `handler`, which takes the specific endpoint its handling), so we could really just create one instance and use it for every route we handle that way instead of creating a new copy for each iteration.